shell/platform/android/io/flutter/plugin/editing/ImeSyncDeferringInsetsCallback.java

```diff
@@ -131,24 +126,24 @@ public WindowInsets onProgress(
       if (!matching) {
         return insets;
       }
+
+      // The IME insets include the height of the navigation bar. If the app isn't laid out behind
+      // the navigation bar, this causes the IME insets will be too large during the animation.
+      // To fix this, we subtract the navigationBars bottom inset if the system UI flags for laying
+      // out behind the navigation bar aren't present.
+      int excludedInsets = 0;
+      int systemUiFlags = view.getWindowSystemUiVisibility();
```

Comment:
getWindowSystemUiVisibility is [deprecated](https://developer.android.com/reference/android/view/WindowManager.LayoutParams#systemUiVisibility) in api 30. @gmackall is working on a similar issue in flutter/flutter/issues/118761. Would you be willing to author a path for post api 30 that uses the windows compat objects that are not depreciated? 

I think based on reading the other code in the PR you only call this class on api 30 and above.

Reply:
As mentioned in my recent update https://github.com/flutter/engine/pull/39391#issuecomment-1512032199 I believe the workaround used in this PR will no longer be required after migrating to the newer APIs, but these changes are quite significant and potentially breaking. I'm not particularly familiar with native Android development so I'll probably be a bit out of my depth, but I'll continue to experiment and create a PR if I get anywhere with it.